src/stores/Frame/RenderConfigStore/RenderConfigStore.ts

```diff
@@ -360,62 +432,148 @@ export class RenderConfigStore {
         }
     };
 
+    /**
+     * Set minimum and maximum values of the scaling.
+     *
+     * @param minVal - The minimum scaling value.
+     * @param maxVal - The maximum scaling value.
+     */
     @action setCustomScale = (minVal: number, maxVal: number) => {
         this.scaleMin[this.stokesIndex] = minVal;
         this.scaleMax[this.stokesIndex] = maxVal;
         this.selectedPercentile[this.stokesIndex] = -1;
         this.updateSiblings();
     };
 
+    /**
+     * Set index of the colormap.
+     *
+     * @param index - The colormap index between -1 and array {@link RenderConfigStore.COLOR_MAPS_ALL} size. The index -1 is the custom color.
+     */
     @action setColorMapIndex = (index: number) => {
-        this.colorMapIndex = clamp(index, 0, RenderConfigStore.COLOR_MAPS_ALL.length - 1);
+        this.colorMapIndex = clamp(index, -1, RenderConfigStore.COLOR_MAPS_ALL.length - 1);
         this.updateSiblings();
     };
 
+    /**
+     * Set the colormap.
+     *
+     * @param colormap - The colormap name in {@link RenderConfigStore.COLOR_MAPS_ALL}.
+     */
     @action setColorMap = (colormap: string) => {
         const index = RenderConfigStore.COLOR_MAPS_ALL.indexOf(colormap);
-        if (index >= 0) {
+        if (colormap === RenderConfigStore.COLOR_MAPS_CUSTOM) {
+            this.setColorMapIndex(RenderConfigStore.CUSTOM_COLOR_MAP_INDEX);
+        } else if (index >= 0 && index < RenderConfigStore.COLOR_MAPS_ALL.length) {
             this.setColorMapIndex(index);
         }
     };
 
+    /**
+     * Set the colormap to be "custom" and its Hex.
```

Comment:
We on longer set the colormap here.
Could we combine `setCustomColorMap` and `setCustomHexEnd`?

Reply:
Yes, it becomes redundant.